src/platform/zephyr/lib/dai.c

```diff
@@ -0,0 +1,308 @@
+		break;
+	case COMP_TRIGGER_PAUSE:
+		zephyr_cmd = DAI_TRIGGER_PAUSE;
+		break;
+	case COMP_TRIGGER_PRE_START:
+	case COMP_TRIGGER_PRE_RELEASE:
+		zephyr_cmd = DAI_TRIGGER_PRE_START;
+		break;
+	default:
+		return -EINVAL;
+	}
+
+	return dai_trigger(dev, direction, zephyr_cmd);
+}
+
+static int dai_set_config_zephyr(struct dai *dai, struct ipc_config_dai *common_config,
+				 void *spec_config)
+{
+	const struct device *dev = dai_get_drvdata(dai);
+	struct sof_ipc_dai_config *sof_cfg;
+	struct dai_config cfg;
+	void *cfg_params;
+	bool is_blob;
+
+	sof_cfg = (struct sof_ipc_dai_config *)spec_config;
+	cfg.dai_index = common_config->dai_index;
+	is_blob = common_config->is_config_blob;
+	cfg.format = sof_cfg->format;
+	cfg.options = sof_cfg->flags;
+
+	switch (common_config->type) {
+	case SOF_DAI_INTEL_SSP:
+		cfg.type = is_blob ? DAI_INTEL_SSP_NHLT : DAI_INTEL_SSP;
+		cfg_params = is_blob ? spec_config : &sof_cfg->ssp; 
+		break;
+	case SOF_DAI_INTEL_ALH:
+		cfg.type = is_blob ? DAI_INTEL_ALH_NHLT : DAI_INTEL_ALH;
+		cfg_params = is_blob ? spec_config : &sof_cfg->alh; 
+		break;
+	default:
+		return -EINVAL;
+	}
+
+	return dai_config_set(dev, &cfg, cfg_params);
+}
+
+static int dai_get_hw_params_zephyr(struct dai *dai, struct sof_ipc_stream_params  *params, int dir)
+{
+	const struct device *dev = dai_get_drvdata(dai);
+	const struct dai_config *cfg;
+
+	cfg = dai_config_get(dev, dir);
+
+	params->rate = cfg->rate;
+	params->buffer_fmt = 0;
+	params->channels = cfg->channels;
+
+	switch (cfg->word_size) {
+	case 16:
+		params->frame_fmt = SOF_IPC_FRAME_S16_LE;
+		break;
+	case 24:
+		params->frame_fmt = SOF_IPC_FRAME_S24_4LE;
+		break;
+	case 32:
+		params->frame_fmt = SOF_IPC_FRAME_S32_LE;
+		break;
+	default:
+		dai_warn(dai, "not supported word size %u", cfg->word_size);
+	}
+
+	return 0;
+}
+
+static int dai_get_handshake_zephyr(struct dai *dai, int direction, int stream_id)
+{
+	const struct device *dev = dai_get_drvdata(dai);
+	const struct dai_properties *props;
+
+	props = dai_get_properties(dev, direction, stream_id);
+
+	return props->dma_hs_id;
+}
+
+static int dai_get_fifo_zephyr(struct dai *dai, int direction, int stream_id)
+{
+	const struct device *dev = dai_get_drvdata(dai);
+	const struct dai_properties *props;
+
+	props = dai_get_properties(dev, direction, stream_id);
+
+	return props->fifo_address;
+}
+
+static uint32_t dai_get_init_delay_ms_zephyr(struct dai *dai)
+{
+	const struct device *dev = dai_get_drvdata(dai);
+	const struct dai_properties *props;
+
+	props = dai_get_properties(dev, 0, 0);
+
+	return props->reg_init_delay;
+}
+
+static int dai_probe_zephyr(struct dai *dai)
+{
+	const struct device *dev = dai_get_drvdata(dai);
+
+	return dai_probe(dev);
+}
+
+static int dai_remove_zephyr(struct dai *dai)
+{
+	const struct device *dev = dai_get_drvdata(dai);
+
+	return dai_remove(dev);
+}
+
+#if CONFIG_INTEL_SSP
+
+static SHARED_DATA struct dai ssp_shared[DAI_NUM_SSP_BASE + DAI_NUM_SSP_EXT];
```

Comment:
should these arrays be converted to Zephyr DT probing in the end? To remove array sizes and just use DT nodes. I guess this can be done later, maybe we aren't ready for that yet